src/celldega/sp/__init__.py

```diff
@@ -0,0 +1,359 @@
+    
+    Returns:
+        A GeoDataFrame containing the bands and mean gene expression.
+    """
+
+    print (f"Complete calculating gene exp. gradient from points to {roi_name} using {band_width} micron rings.")
+
+
+    # Merge all bands into a single GeoDataFrame
+    gdf_bands = create_concentric_rings(gdf_polygon, num_bands, band_width)
+    
+    # Spatial join: Assign each cell to the closest buffer
+    gdf_join_all = gdf_points.sjoin(gdf_bands, how="left", predicate="within").drop(
+        columns=['index_right'], errors='ignore')
+    
+    # Compute mean expression per band
+    gdf_join_all["band"] = (
+        gdf_join_all[f"distance_to_polygon_{roi_name}"] // band_width
+    ).astype(int)
+    
+    # Compute mean expression for each gene
+    for gene in gene_list:
+        band_avg_expression = gdf_join_all.groupby("band")[gene].mean().reset_index()
+        band_avg_expression.columns = ["band", f"{gene}_mean"]
+        gdf_join_all = gdf_join_all.merge(band_avg_expression, on="band")
+
+    return gdf_join_all, gdf_bands
+
+def plot_gene_gradient_to_roi(
+    gdf_join_all: gpd.GeoDataFrame,
+    gdf_polygon: gpd.GeoDataFrame,
+    roi_name: str,
+    gene_list: list,
+    markersize: float = 1
+):
+    """
+    Plot gene expression gradients relative to the region of interest.
+    
+    Args:
+        gdf_join_all: GeoDataFrame containing bands and gene expression data.
+        gdf_polygon: GeoDataFrame containing the target roi.
+        roi_name: Name of the region of interest.
+        gene_list: List of genes to plot gradients for.
+        markersize: Size of the points in the plot.
+    """
+
+    print (f"Plotting gene expression gradient from points to ROI {roi_name}...")
+
+    for gene in gene_list:
+        # Plot results
+        fig, ax = plt.subplots(figsize=(20, 8))
+        
+        # Plot points outside the roi with gradient color
+        gdf_join_all.plot(
+            ax=ax,
+            column=f"{gene}_mean",
+            aspect=1,
+            cmap="coolwarm",
+            legend=True,
+            alpha=0.7,
+            markersize=markersize,
+        )
+
+        # Plot the roi as a filled area with label
+        gdf_polygon.plot(
+            ax=ax, color="gray", aspect=1, alpha=0.5, edgecolor="black", linewidth=2
+        )
+
+        # Annotate the polygon with the ROI name at its centroid
+        centroid = gdf_polygon.geometry.iloc[0].centroid
+        ax.annotate(
+            text=roi_name,
+            xy=(centroid.x, centroid.y),
+            ha="center",
+            va="center",
+            fontsize=12,
+            color="black",
+        )         
+
+        ax.invert_yaxis()
+        plt.title(f"Gradient of {gene} Expression per Cell Along Distance to ROI")
+        plt.xlabel("x (micron)")
+        plt.ylabel("y (micron)")
+        plt.tight_layout()
+        plt.show()
+
+
+def create_band_pixel(gdf_micron, gdf_pixel, roi_name, n_rings, band_width, json_fname):
```

Comment:
can we hide this from the user facing API (add underscore to name)?

Reply:
fixed